crypto/src/crypto_test.rs

```diff
@@ -1,39 +1,34 @@
-use std::path::Path;
+use std::{fs, path::Path};
 
-use bn254::ECDSA;
-use ed25519_dalek::{Keypair as Ed25519DalekKeyPair, SecretKey, Signature, Signer};
+use bn254::{PrivateKey, ECDSA};
+use ed25519_dalek::{Keypair, Signature, Signer};
 use rand::rngs::OsRng;
-use tokio::fs::write;
 
-use crate::{derive_bn254_key_pair, derive_ed25519_key_pair, Bn254KeyPair, Ed25519KeyPair};
+use super::*;
 
 const TEST_SK_PATH: &str = "./seda_test_sk";
-async fn generate_test_sk() {
+fn generate_test_sk() {
     if !Path::new(&TEST_SK_PATH).exists() {
-        let mut csprng = OsRng {};
-        let sk = SecretKey::generate(&mut csprng);
-        write(TEST_SK_PATH, hex::encode(&sk))
-            .await
-            .expect("Unable to write secret key");
+        let sk = PrivateKey::random(&mut OsRng);
+        fs::write(TEST_SK_PATH, hex::encode(sk.to_bytes().unwrap())).expect("Unable to write secret key");
     }
 }
 
-#[tokio::test]
-async fn generate_bn254_pair() {
-    generate_test_sk().await;
-    let bn_pair: Bn254KeyPair = derive_bn254_key_pair(TEST_SK_PATH, 1).expect("Couldn't derive bn254 key pair");
+#[test]
+fn generate_bn254_pair() {
+    generate_test_sk();
+    let bn_pair = KeyPair::derive(TEST_SK_PATH, 1).expect("Couldn't derive bn254 key pair");
```

Comment:
Should this `KeyPair` be the one from the crypto module? (and not from ed25519_dalek)
```suggestion
    let bn_pair = KeyPair::derive(TEST_SK_PATH, 1).expect("Couldn't derive bn254 key pair");
```

Reply:
It is the one from ours! Notice `KeyPair`(ours) vs. `Keypair` (`ed25519_dalek`).